p2p/libp2p/discovery/hostDecorator.go

```diff
@@ -0,0 +1,182 @@
+package discovery
+
+import (
+	"context"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/libp2p/go-libp2p-core/connmgr"
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	protocol "github.com/libp2p/go-libp2p-core/protocol"
+	peerstore "github.com/libp2p/go-libp2p-peerstore"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+const (
+	hardCPSLimitLow  = 2
+	hardCPSLimitHigh = 100
+)
+
+type hostDecorator struct {
+	h            host.Host
+	acceptedChan chan struct{}
+	resumeChan   chan struct{}
+	timeout      time.Duration
+}
+
+// NewHostDecorator creats a new decorator around an existing host (h) that will olny allow
+// cps connections per second and drop the connection request upon timeout expiration
+func NewHostDecorator(h host.Host, ctx context.Context, cps uint32, timeout time.Duration) (*hostDecorator, error) {
+	if h == nil {
+		return nil, p2p.ErrNilHost
+	}
+
+	if ctx == nil {
+		return nil, p2p.ErrNilContext
+	}
+
+	if timeout < 0 {
+		return nil, p2p.ErrInvalidDurationProvided
+	}
+
+	if cps < hardCPSLimitLow {
+		cps = hardCPSLimitLow
```

Comment:
this should return error.
Also on line 50

Reply:
i prefer just to keep it in check

Reply:
You should stick to the rules enforced by the whole team.

Reply:
Yes, the convention was that if a "config" is not ok, we should return an error. If the hardCPSLimitLow was changed and the "config" was not, the "app" should error. This way if it works, we know that the cps was correct.

Reply:
Also some tests should be added after.